spec/components/search_spec.rb

```diff
@@ -802,4 +802,32 @@ def search
     end
   end
 
+  context 'pagination' do
+    let(:number_of_results) { 2 }
+    before do
+      5.times { Fabricate(:post) }
+      Search.stubs(:per_filter).returns(number_of_results)
+    end
+
+    it 'returns more results flag' do
+      results = Search.execute('hello', type_filter: 'topic')
+      results2 = Search.execute('hello', type_filter: 'topic', page: 2)
+
+      expect(results.posts.length).to eq(number_of_results)
+      expect(results.more_full_page_results).to eq(true)
+      expect(results2.posts.length).to eq(number_of_results)
```

Comment:
Add check here that post ids aren't same as the first page. This test will pass if the page param is ignored I think.